mongo/integration/client_side_encryption_prose_test.go

```diff
@@ -1035,6 +1036,50 @@ func TestClientSideEncryptionProse(t *testing.T) {
 			})
 		}
 	})
+
+	// These tests only run when a KMS mock server is running on localhost:8000.
+	mt.RunOpts("kms tls tests", noClientOpts, func(mt *mtest.T) {
+		testcases := []struct {
+			name       string
+			envValue   string
+			errMessage string
+		}{
+			{
+				"invalid certificate",
+				"INVALID_CERT",
+				"expired",
+			},
+			{
+				"invalid hostname",
+				"INVALID_HOSTNAME",
+				"SANs",
+			},
+		}
+
+		for _, tc := range testcases {
+			mt.Run(tc.name, func(mt *mtest.T) {
+				// Only run test if correct KMS mock server is running.
+				if os.Getenv("KMS_TLS_TESTCASE") == tc.envValue {
+					ceo := options.ClientEncryption().
+						SetKmsProviders(fullKmsProvidersMap).
+						SetKeyVaultNamespace(kvNamespace)
+					cpt := setup(mt, nil, nil, ceo)
```

Comment:
This is missing a `defer cpt.teardown(mt)` call.

Reply:
Good catch! Added.